src/Compilers/Core/Portable/CodeGen/LocalSlotManager.cs

```diff
@@ -64,6 +64,23 @@ public override bool Equals(object? obj)
         // maps local identities to locals.
         private Dictionary<ILocalSymbolInternal, LocalDefinition>? _localMap;
 
+        // The lowered tree might define the same local symbol
+        // in multiple sequences that are part of one expression, for example:
+        //
+        //   call:
+        //     arguments:
+        //       sequence:
+        //         locals: LoweringTemp.1
+        //         ...
+        //       sequence:
+        //         locals: LoweringTemp.1
+        //         ...
+        //
+        // At the same time, these expression locals might be lifted to the containing block
+        // (to avoid reusing them if they might be captured by a ref struct).
+        // Then we use this map to keep track of the redeclared locals.
+        private Dictionary<LocalDefinition, LocalDefinition>? _redeclaredLocals;
```

Comment:
>private Dictionary<LocalDefinition, LocalDefinition>? _redeclaredLocals; [](http://example.com/codeflow?start=8&length=72)

It is not obvious what do map to what here. Could you, please, elaborate and add a comment?

Reply:
It maps from the old LocalDefinition to the new LocalDefinition (which is redeclaring/shadowing the old one). Does that make sense?

I will try to explain it in the comment. Thanks.